src/components/security_manager/test/CMakeLists.txt

```diff
@@ -28,31 +28,24 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-if (BUILD_TESTS)
-  
+include(${CMAKE_SOURCE_DIR}/tools/cmake/helpers/sources.cmake)
+
 include_directories(
   ${GMOCK_INCLUDE_DIRECTORY}
   ${COMPONENTS_DIR}/protocol_handler/test/include
   ${COMPONENTS_DIR}/transport_manager/test/include
-  ${COMPONENTS_DIR}/security_manager/include
-  ${COMPONENTS_DIR}/resumption/include
   ${JSONCPP_INCLUDE_DIRECTORY}
 )
 
-set(SOURCES
-  ${COMPONENTS_DIR}/security_manager/test/crypto_manager_impl_test.cc
-  ${COMPONENTS_DIR}/security_manager/test/security_manager_test.cc
-  ${COMPONENTS_DIR}/security_manager/test/security_query_test.cc
-  ${COMPONENTS_DIR}/security_manager/test/security_query_matcher.cc
-  ${COMPONENTS_DIR}/security_manager/test/ssl_context_test.cc
-  ${COMPONENTS_DIR}/security_manager/test/ssl_certificate_handshake_test.cc
- )
+collect_sources(SOURCES "${CMAKE_CURRENT_SOURCE_DIR}")
 
 set(LIBRARIES
   gmock
-  ${SecurityManagerLibrary}
+  security_manager
 )
 
+create_test(security_manager_test "${SOURCES}" "${LIBRARIES}")
+
 set(CERT_LIST 
```

Comment:
@anosach-luxoft 
Maybe it worth to use something similar to collect_sources(), i.e. for all possible *.pem files in case naming will be changed in generator.